apps/whale-api/src/module.api/legacy.controller.ts

```diff
@@ -0,0 +1,334 @@
+import { Controller, Get, Inject, Logger, Query } from '@nestjs/common'
+import { BlockMapper } from '../module.model/block'
+import { SupportedNetwork } from '../../../legacy-api/src/pipes/NetworkValidationPipe'
+import { parseHeight } from './block.controller'
+import { BlockTxn, encodeBase64 } from '../../../legacy-api/src/controllers/PoolPairController'
+import { Block } from '@defichain/whale-api-client/dist/api/blocks'
+import { requireValue } from './stats.controller'
+import { Transaction, TransactionVout } from '@defichain/whale-api-client/dist/api/transactions'
+import {
+  CCompositeSwap,
+  CompositeSwap,
+  CPoolSwap,
+  OP_DEFI_TX,
+  PoolSwap,
+  toOPCodes
+} from '@defichain/jellyfish-transaction'
+import { fromScript } from '@defichain/jellyfish-address'
+import { AccountHistory } from '@defichain/jellyfish-api-core/dist/category/account'
+import { SmartBuffer } from 'smart-buffer'
+import { BigNumber } from 'bignumber.js'
+import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
+import { TransactionVoutMapper } from '../module.model/transaction.vout'
+import { TransactionMapper } from '../module.model/transaction'
+import { SimpleCache } from '../../../legacy-api/src/cache/SimpleCache'
+import { NetworkName } from '@defichain/jellyfish-network'
+import { ApiRawResponse } from './_core/api.response'
+
+/**
+ * Specifically for bypassing response interceptor
+ */
+class LegacyApiRawResponse extends ApiRawResponse {
+  readonly data: LegacySubgraphSwapsResponse['data']
+  readonly page: LegacySubgraphSwapsResponse['page']
+
+  constructor (swaps: LegacySubgraphSwapsResponse) {
+    super()
+    this.data = swaps.data
+    this.page = swaps.page
+  }
+}
+
+@Controller('/legacy')
+export class LegacyController {
+  private readonly logger = new Logger(LegacyController.name)
+  constructor (
+    protected readonly rpcClient: JsonRpcClient,
+    protected readonly blockMapper: BlockMapper,
+    protected readonly transactionMapper: TransactionMapper,
+    protected readonly transactionVoutMapper: TransactionVoutMapper,
+    private readonly cache: SimpleCache,
```

Comment:
I think we should remove the cache in `whale-api` and let it hit defid for the account history data? Otherwise, it would compete with whale-api cache which is critically more important.

Reply:
Unless we can separate the cache layer since they compete with the same cache provider size.

Reply:
Currently, on `poolpair.controller.ts`, we have a simple implementation (also account history) without caching and it performs fine since everything is enclaved into a single `ocean: whale + defid` group within the local network hence the fast lookup.

Reply:
will remove and verify the performance. I think it's fine since we'll also be limiting to 20 swaps each query

Reply:
added cache back as performance is unacceptable without it